### ISSUE TITLE
ProjectTo is returning wrong number of objects(null) while using it in unit tests.

### ISSUE THREAD

**Issue by melikpehlivanov** [AutoMapper/AutoMapper#2831]
Hello, hope you're having a great time! I came across at very strange problem.

When I unit test a given service and for example let's suppose I have some data in my inmemory database the following things happen:
1. If I use ProjectTo the entity which is returned is null although I have entity in my inmemory database so of course my unit test fails.
2. If I use this.mapper.Map<destinationType>() the entity which is returned is correct in the unit test and the unit test passes successfully but when a controller calls this method the result which is returned is not correct because all navigation properties in it are not mapped correctly. 

Please review this gist for more information - https://gist.github.com/melikpehlivanov/e799712004826c0e3e4ab78076a2ef05

In conclusion, ProjectTo doesn't work with unit tests and I think there's a bug, if not, please tell me what should I do.

**Comment by jbogard** [comment]
My only guess is the in-memory database can’t handle projections. Try
rewriting without AutoMapper for ProjectTo using only Select LINQ.

**Comment by melikpehlivanov** [comment]
Hi there, if I use Select LINQ it works but I've seen projects where ProjectTo works with inmemorydatabase too
